MapboxNavigationTests/RouteControllerSnapshotTests.swift

```diff
@@ -22,13 +22,20 @@ class RouteControllerSnapshotTests: FBSnapshotTestCase {
     }
     
     func testRouteSnappingOvershooting() {
-        let route = Fixture.routesFromMatches(at: "sthlm-double-back")![0]
+        let routeExpectation = expectation(description: "Gets a matching route")
+        var route: Route!
+        Fixture.getRoutesFromMatches(at: "sthlm-double-back") { (waypoints, routes, error) in
+            route = routes?[0]
+            routeExpectation.fulfill()
+        }
+        wait(for: [routeExpectation], timeout: 1)
+        guard route != nil else { return }
```

Comment:
Is this guard really needed? If it fails, should the test fail too?

Reply:
The test will already fail if there’s no `route`, because the expectation will fail to be fulfilled. I prefer to minimize cascading test failures to avoid obscuring the original failure.